api/Dockerfile

```diff
@@ -82,7 +86,11 @@ RUN set -eux; \
 	composer install --prefer-dist --no-dev --no-autoloader --no-scripts --no-progress --no-suggest; \
 	composer clear-cache
 
-COPY . ./
+# copy only specifically what we need
```

Comment:
It's a docker best practice to let the .dockerignore file configure exclusion: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#exclude-with-dockerignore

Reply:
No, that's a misunderstanding of what the `.dockerignore` file is for. We should only ever put in `.dockerignore` what will **never** be copied into the Docker image.

For example, if we add `docker/` in the `api/.dockerignore` file, the image wouldn't build.

Reply:
@vincentchalamon What are you disagreeing with? I'm merely pointing out facts. The `.dockerignore` file excludes from the Docker build context. If we keep to `COPY . ./`, the `docker` directory will always be copied into the image, which is of course undesirable.